### ISSUE TITLE
Can electron-boilerplate work with Electron v5 and up?

### ISSUE THREAD

**Issue by guanxun14** [szwacz/electron-boilerplate#308]
I was unable to adapt my project (which was set up with `electron-boilerplate` 6.0.0, that uses gulp) to successfully build with electron v5 or v6, and so seems like I am capped at 4.2.9 for now.

With the new webpack-based versions however, you also recently seemed to run into [issues](https://github.com/szwacz/electron-boilerplate/issues/306#issuecomment-516561862) trying to update electron in this repo. Wondering if there is a path forward for this / workaround?

BTW, thanks for your time and work on this repo — has helped thousands of Electron projects off the ground (including mine)! :)

**Comment by black-snow** [comment]
@guanxun14 what issues did you encounter?

**Comment by guanxun14** [comment]
Was able to get `npm start` running after updating node to v12 and gulp to v4 (because of some promise-related error, if I recall correctly), and making the required syntax changes in the gulp task files. 

However with the build command `npm run release`, it simply failed with a cryptic `sh: build: command not found` at quite an early stage of the build process, and not much else to go on. Can attempt to retrace steps/errors in full if you'd find it helpful.

Not even sure if I'm approaching this right. Were you able to update to v5/6 without (blocking) issues?

**Comment by black-snow** [comment]
Personally I haven't done any electron related projects lately.
It'd, of course, help if you'd trace down where things go wrong. You might be able to fix things yourself. I wonder where the `sh`ell comes in.

**Comment by szwacz** [comment]
Hi! The main problem is that I don't have electron-based project in active development as well. So I'm just updating the boilerplate, what is much easier than migrating living and maintained app. I don't even have gulp-building project, so don't know what's going on in that ecosystem now.

Hint: `build` was command for launching `electron-builder` and that project changed it to something more descriptive right now, so this might be simple name change to do and everything will be working.

That being said, the build tools are quite interchangeable (it's javascript after all), so you can try to do small R&D and just copy your source code to newest version of boilerplate and give it one hour to see how much code change and debugging it will need to work (estimate the full migration time).

The electron ecosystem is quite mature now, and there aren't much changes, so keeping evergreen shouldn't be that difficult.
Regarding electron v5, I had problem only with Spectron (end to end testing engine), which wasn't starting, that halted me and I reverted one version back. The rest was working fine.

**Comment by guanxun14** [comment]
Perhaps reveals my inexperience/incompetence, but I'm happy to report that I managed to get things working! :)

> Hint: `build` was command for launching `electron-builder` and that project changed it to something more descriptive right now, so this might be simple name change to do and everything will be working.

This got me looking in the right place. Updated the `"release": "build",` line in `package.json` to `"release": "electron-builder build",`, and it worked just fine.

I also found [these](https://fettblog.eu/gulp-4-parallel-and-series/) [links](https://www.liquidlight.co.uk/blog/how-do-i-update-to-gulp-4/) helpful for migrating to Gulp v4, so I shall just leave them here too in case others find it helpful.

Thank you for the comments and I shall mark this issue as closed. Cheers!
